src/workerd/api/global-scope.h

```diff
@@ -537,17 +537,15 @@ class ServiceWorkerGlobalScope: public WorkerGlobalScope {
       jsg::Function<void(jsg::Arguments<jsg::Value>)> function,
       jsg::Optional<double> msDelay,
       jsg::Arguments<jsg::Value> args);
-  void clearTimeout(kj::Maybe<TimeoutId::NumberType> timeoutId);
+  void clearTimeout(jsg::Lock& js, kj::Maybe<jsg::JsValue> timeoutId);
```

Comment:
This could probably be `jsg::JsNumber`? That might avoid the need for the type override.